python/ambassador/ir/irhost.py

```diff
@@ -473,7 +478,41 @@ def matches_httpgroup(self, group: "IRHTTPMappingGroup") -> bool:
                 sel_match,
             )
 
-        return host_match or sel_match
+        groupName = group.get("name") or "None"
+
+        # The synthetic Mappings for diagnostics, readiness, and liveness probes always match all Hosts.
+        # They can all still be disabled if desired via the Ambassador Module resource
+        if groupName in [
+            "GROUP: internal_readiness_probe_mapping",
+            "GROUP: internal_liveness_probe_mapping",
+            "GROUP: internal_diagnostics_probe_mapping",
+        ]:
+            return True
+
+        # Ambassador (2.0-2.3) & (3.0-3.1) consider a match on a single label as a "good enough" match.
+        # In versions 2.5+ and 3.2+ _ALL_ labels in a selector must be present for it to be considered a match.
+        # DISABLE_STRICT_LABEL_SELECTORS provides a way to restore the old unintended loose matching behaviour
+        # in the event that it is desired. The ability to disable strict label matching will be removed in a future version.
+        disable_strict_selectors = parse_bool(
```

Comment:
Looks like we duplicate this twice, maybe we just move it into a util function so the comment/default are maintained in a single place?